### PR TITLE
Depend on external libjansson and fix for AVRO-1691

### PR DESCRIPTION
**Pull request by edenhill**
Fix for AVRO-1691:
Allow JSON string-field primitive schema definitions
https://issues.apache.org/jira/browse/AVRO-1691

This fix requires a newer version of libjansson (>=2.3) that features the JSON_DECODE_ANY decoder flag. Since libjansson is widely available in distributions nowadays the old bundled copy was removed in favour of depending on libjansson as an external library.

### CODE REVIEW

**Comment by busbey** [comment]
There should be a JIRA for each commit. Could you add a jira for the library being external + updated?

File: lang/c/FindJansson.cmake [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# - Try to find Jansson
```
Comment (busbey) [inline lang/c/FindJansson.cmake]: this file needs an ASF header

File: lang/c/FindJansson.cmake [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# Once done this will define
+#
+#  JANSSON_FOUND - system has Jansson
+#  JANSSON_INCLUDE_DIRS - the Jansson include directory
+#  JANSSON_LIBRARIES - Link these to use Jansson
+#
+#  Copyright (c) 2011 Lee Hambley <lee.hambley@gmail.com>
```
Comment (busbey) [inline lang/c/FindJansson.cmake]: LICENSE/NOTICE files need to be updated for the inclusion of this file

**Comment by busbey** [comment]
INSTALL file still claims no external dependencies.

**Comment by busbey** [comment]
LICENSE/NOTICE files need to be updated for hte removal of Jansson from the source tree.

**Comment by busbey** [comment]
INSTALL needs to be updated with a pointer on installing Jansson.

**Comment by busbey** [comment]
Docker build needs to be updated to install Jansson.

**Comment by ept** [comment]
@busbey I've made those updates and posted a patch for removing Jansson from the source tree: https://issues.apache.org/jira/browse/AVRO-1769

**Pull request closed without merging**